### ISSUE TITLE
`https` misalignment with Node

### ISSUE THREAD

**Issue by fucksophie** [oven-sh/bun#2388]
### What version of Bun is running?

Bun 0.5.7

### What platform is your computer?

Linux 5.10.0-16-amd64 x86_64

### What steps can reproduce the bug?

Run `bun add dockerode` and run this code:
```
import Dockerode from "dockerode";

Dockerode;
```

### What is the expected behavior?

The module imports properly with no errors

### What do you see instead?

```
15 | for (var protocol in protocols) {
16 |   var h = function() {};
17 |   h.prototype = protocols[protocol];
18 |   h = new h();
19 |   console.log(h)
20 |   h.request = function(h) {
      ^
TypeError: Attempted to assign to readonly property.
      at /example/node_modules/docker-modem/lib/http.js:20:2
      at /example/node_modules/docker-modem/lib/modem.js:2:2
      at /example/node_modules/dockerode/lib/docker.js:2:2
      at /example/index.ts:3:0
```

### Additional information

Error stems from this piece of code: https://github.com/apocas/docker-modem/blob/d5c20c5581f1aa99bd89a234733365d9b58ddef7/lib/http.js#L20

Issue: https://nodejs.org/dist/latest-v19.x/docs/api/https.html#httpsrequesturl-options-callback should NOT be readonly, because it isn't readonly in Node.

Node example (this uses https.get but https.request, https.post, https.. have this issue):
```js
import https from "https";

https.original = https.get;

https.get = (url, callback) => {
  console.log("Overwritten!" , url, callback)
  return https.original(url, callback)
}

https.get('https://google.com', res => {
  let data = [];

  res.on('data', chunk => {
    data.push(chunk);
  });

  res.on('end', () => {
    console.log('Response ended: ');
    console.log(data.join(""))
  });
})

```

**Comment by yourfriendoss** [comment]
Seems like the `readonly` here is unecessary: https://github.com/oven-sh/bun/blob/4c3879814289f79d81aef30e55c4bad9145c2af8/packages/bun-types/http.d.ts#L572

**Comment by fucksophie** [comment]
Issue fixed in commit [27f5012f50b998fec5b45531de2439a11a72eaa2](https://github.com/oven-sh/bun/commit/27f5012f50b998fec5b45531de2439a11a72eaa2)
